### PR TITLE
Require secrets to be present

### PR DESCRIPTION
**Pull request by julienduchesne**
Adds a `DRONE_RUNNER_SECRETS_REQUIRED` env variable that instructs Drone to crash if secrets are not found on their provider
If this env variable is not set, current behavior remains which is to create the env variable with an empty value, making the feedback to the user much less clear

image

### CODE REVIEW

File: engine/engine.go [hunk 1]
```diff
@@ -6,8 +6,11 @@ package engine
 
 import (
 	"context"
+	goerrors "errors"
```
Comment (inkel) [inline engine/engine.go:9]: Is this renaming still needed?

**Review by inkel — APPROVED** [review]
LGTM! Great idea and neat implementation.

File: engine/engine.go (hunk 1 above)
Comment (julienduchesne) [reply engine/engine.go:9]: `errors` is used for an internal package a few lines below

**Comment by marko-gacesa** [comment]
@julienduchesne thank you for your effort. However, we believe that this might not be the best approach. You added this as part of `engine` implementation, while a better place would be in `compiler` package. Also, it might be better to avoid another environment variable and extend yaml for this. For example we might say in yaml:

```
  settings:
    password:
      from_secret: docker_password
      required: true
```

**Comment by tphoney** [comment]
If you implement this change as suggested by @marko-gacesa in the compiler package, we will have a look. 

Thanks for your patience.

**Comment by julienduchesne** [comment]
> If you implement this change as suggested by @marko-gacesa in the compiler package, we will have a look.
> 
> Thanks for your patience.

Just looked at this another time. The reason I didn't implement it in the `compiler` package is that the `compiler` package has no way to return an error and the `Compile` method comes from the `runner-go` package. Should we change it to support errors over there then implement errors in the various runners?

**Pull request closed without merging**